app/src/main/database/index.ts

```diff
@@ -252,4 +256,88 @@ export class DB {
       });
     });
   }
+
+  // Helper function for truthy values
+  private isTruthy(value: unknown): boolean {
+    if (typeof value === "boolean") return value;
+    if (typeof value === "number") return value !== 0;
+    if (typeof value === "string")
+      return value.toLowerCase() === "true" || value === "1";
+    return Boolean(value);
+  }
+
+  // UI query methods from database-stuff branch
+  getSources(): Source[] {
+    if (!this.db) {
+      throw new Error("Database is not open");
+    }
+
+    // Select sources
+    const stmt = this.db.prepare(`
+      SELECT
+        uuid,
+        data,
+        is_seen,
+        has_attachment,
+        show_message_preview,
+        message_preview
+      FROM sources
+    `);
+
+    const rows = stmt.all() as Array<SourceRow>;
+
+    return rows.map((row) => {
+      const data = JSON.parse(row.data) as SourceMetadata;
+      return {
+        uuid: row.uuid,
+        data,
+        isRead: this.isTruthy(row.is_seen),
```

Comment:
Should we rename this it `isSeen` too?

Reply:
Yeah, I was just trying to keep the scope of all the things that were being rename down a bit; if it's okay I'd rather do that as a follow-up once all the UI stuff has landed too.